Node version 14 for CI

dartamonov-vertex opened the pull request.

yoyo837 commented:
CI failed.

dartamonov-vertex commented:
Something must be off with CI. This PR failed with 0 changes to the source code.

afc163 commented on tsconfig.json:
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+      "allowSyntheticDefaultImports": true,
+      "declaration": true,
+      "module": "esnext",
+      "target": "esnext",
+      "moduleResolution": "node",
+      "jsx": "react",
+      "skipLibCheck": true
+    },
+    "include": ["./src", "./typings/"],
+    "typings": "./typings/index.d.ts",
+    "exclude": [
+      "node_modules",
+      "build",
+      "scripts",
+      "acceptance-tests",
+      "webpack",
+      "jest",
+      "src/setupTests.ts",
+      "tslint:latest",
+      "tslint-config-prettier"
+    ]
+}
```
```suggestion
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "declaration": true,
    "module": "esnext",
    "target": "esnext",
    "moduleResolution": "node",
    "jsx": "react",
    "skipLibCheck": true
  },
  "include": ["./src", "./typings/"],
  "typings": "./typings/index.d.ts",
  "exclude": [
    "node_modules",
    "build",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "src/setupTests.ts",
    "tslint:latest",
    "tslint-config-prettier"
  ]
}
```

afc163 commented on src/Dropdown.tsx:
```diff
@@ -1,8 +1,15 @@
+  ActionType,
+} from 'rc-trigger/lib/interface';
 import Placements from './placements';
+import useAccessibility from './hooks/useAccessibility';
```
Does it belong to #182 which should not be commited here?

dartamonov-vertex replied:
Updated with the push below

dartamonov-vertex commented:
@afc163 @MadCcc This PR didn't add any new functional code. It was pretty much a test of the CI you introduced. If you don't want to use this PR, can you please create your own and test if it pass your CI.
You are welcome to use the fixes introduced in this PR if you need them:
* Added tsconfig.json to override father default config
* Fixed lint errors in src/Dropdown.tsx
* Updated node version to 14 in CI, because node 12 will be end of life'd in April

The last error on this PR - https://github.com/react-component/dropdown/pull/191/files#diff-c3c9c10a6e6fab8c821dc6ccbb2ffd57c5750035b6cfc76fed0afe78ef98121eR97 is related to the renamed variable. That line is not a new line. It just renames a variable to fix the lint issue.

afc163 commented:
<img width="833" alt="图片" src="https://user-images.githubusercontent.com/507615/150714011-6a039e3c-d48f-4610-9fea-83e42fce4305.png">

MadCcc commented:
I will work on it.

MadCcc commented:
updated https://github.com/react-component/dropdown/pull/192
